File: 1.1/src/cz/mg/vulkan/PFNvkFreeMemory.java
package cz.mg.vulkan;

public class PFNvkFreeMemory extends VkFunctionPointer {
    public PFNvkFreeMemory() {
    }

    protected PFNvkFreeMemory(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkFreeMemory(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkFreeMemory(long value) {
        setValue(value);
    }

    public PFNvkFreeMemory(VkInstance instance) {
        super(instance, new VkString("vkFreeMemory"));
    }

    public void call(VkDevice device, VkDeviceMemory memory, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long memory, long pAllocator);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkFreeMemory.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkFreeMemory_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong memory, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkFreeMemory f = (PFN_vkFreeMemory)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

